### ISSUE TITLE
Server crash (ticking entity) on item clear

### ISSUE THREAD

**Issue by sprocketaudio** [Darkere/CrashUtilities#34]
Hello, thanks for fixing item clear - however there seems to now be a server crash after it runs. This has happened just twice this morning and from reports of my users they have said the server seems to crash following an item clear.

Here is the log file from the moment it starts to run..

```
[21Feb2023 07:42:15.412] [Timer-5/INFO] [net.minecraft.server.MinecraftServer/]: [=== ITEMCLEAR IN 20 SECONDS ===]
[21Feb2023 07:42:15.412] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Showing new title for Legend - GeekRiot
[21Feb2023 07:42:25.411] [Timer-3/INFO] [net.minecraft.server.MinecraftServer/]: [=== ITEMCLEAR IN 10 SECONDS ===]
[21Feb2023 07:42:30.411] [Timer-2/INFO] [net.minecraft.server.MinecraftServer/]: [=== ITEMCLEAR IN 5 SECONDS ===]
[21Feb2023 07:42:35.418] [Timer-4/INFO] [net.minecraft.server.MinecraftServer/]: 1573 Items cleared
[21Feb2023 07:42:35.516] [Server thread/ERROR] [net.minecraft.server.MinecraftServer/]: Encountered an unexpected exception
net.minecraft.ReportedException: Ticking entity
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:870) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:292) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1013) ~[?:?]
	at java.util.ArrayList$Itr.next(ArrayList.java:967) ~[?:?]
	at com.google.common.collect.Iterators$1.next(Iterators.java:146) ~[guava-31.0.1-jre.jar%2372!/:?]
	at net.minecraft.world.level.entity.EntitySection.m_188352_(EntitySection.java:33) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.entity.EntitySectionStorage.m_188365_(EntitySectionStorage.java:120) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.entity.EntitySectionStorage.m_188362_(EntitySectionStorage.java:65) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.entity.EntitySectionStorage.m_156890_(EntitySectionStorage.java:120) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.entity.LevelEntityGetterAdapter.m_142232_(LevelEntityGetterAdapter.java:42) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.Level.m_6249_(Level.java:588) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.EntityGetter.m_183134_(EntityGetter.java:62) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.CommonLevelAccessor.m_183134_(CommonLevelAccessor.java:24) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.CollisionGetter.m_45756_(CollisionGetter.java:57) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.entity.item.ItemEntity.m_8119_(ItemEntity.java:122) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.level.ServerLevel.track(ServerLevel.java:4740) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.level.ServerLevel.redirect$bfe000$onTickNonPassenger(ServerLevel.java:4756) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:658) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.Level.m_46653_(Level.java:457) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:323) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:303) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[server-1.19.2-20220805.130853-srg.jar%23741!/:?]
	... 5 more
```

And here is the crash report
[crash-2023-02-21_07.42.35-server.txt](https://github.com/Darkere/CrashUtilities/files/10792236/crash-2023-02-21_07.42.35-server.txt)

Further information - I found this is a base with a mob farm, and an absorbtion hopper (which is only sucking 1 item at a time it seems).. I do not know for sure but could the hopper be taking the item from the world before the clear runs? Or some other similar conflict?

**Comment by Darkere** [comment]
Yep, just checked my fault. Will fix soon.

**Comment by Darkere** [comment]
fixed in 6.2 and 7.1
